test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/ObjectManagement/ObjectManagementServiceTests.cs

```diff
@@ -0,0 +1,171 @@
+            await CheckGeneratedScript(testDb.DatabaseName, "CREATE TABLE [dbo].[RenamingTable]");
+
+            await SqlTestDb.DropDatabase(testDb.DatabaseName);
+        }
+
+        [Test]
+        public async Task TestRenameColumn()
+        {
+            await objectManagementService.HandleRenameRequest(this.InitRequestParams("RenameColumn", String.Format("Server/Database[@Name='{0}']/Table[@Name='testTable1_RenamingTable' and @Schema='dbo']/Column[@Name='C1']", testDb.DatabaseName)), requestContextMock.Object);
+            Thread.Sleep(2000);
+            requestContextMock.Verify(x => x.SendResult(It.Is<bool>(r => VerifySendedFeedback(r, true))));
+            await CheckGeneratedScript(testDb.DatabaseName, "[RenameColumn] [int] NULL");
+
+            await SqlTestDb.DropDatabase(testDb.DatabaseName);
+        }
+
+        [Test]
+        public async Task TestRenameColumnNotExisting()
+        {
+            Assert.ThrowsAsync<FailedOperationException>(async () =>
+            {
+                await objectManagementService.HandleRenameRequest(this.InitRequestParams("RenameColumn", String.Format("Server/Database[@Name='{0}']/Table[@Name='testTable1_RenamingTable' and @Schema='dbo']/Column[@Name='C1_NOT']", testDb.DatabaseName)), requestContextMock.Object);
+                Thread.Sleep(2000);
+            });
+
+            await SqlTestDb.DropDatabase(testDb.DatabaseName);
+        }
+
+        [Test]
+        public async Task TestRenameTableNotExisting()
+        {
+            Assert.ThrowsAsync<FailedOperationException>(async () =>
+            {
+                await objectManagementService.HandleRenameRequest(this.InitRequestParams("RenamingTable", String.Format("Server/Database[@Name='{0}']/Table[@Name='testTable1_Not' and @Schema='dbo']", testDb.DatabaseName)), requestContextMock.Object);
+                Thread.Sleep(2000);
+            });
+            await SqlTestDb.DropDatabase(testDb.DatabaseName);
+        }
+
+        [Test]
+        public async Task TestConnectionNotFound()
+        {
+            var testRenameRequestParams = new RenameRequestParams
+            {
+                NewName = "RenamingTable",
+                ConnectionUri = "NOT_EXISTING",
+                ObjectUrn = String.Format("Server/Database[@Name='{0}']/Table[@Name='testTable1_Not' and @Schema='dbo']", testDb.DatabaseName),
+            };
+            Assert.ThrowsAsync<Exception>(async () =>
+            {
+                await objectManagementService.HandleRenameRequest(testRenameRequestParams, requestContextMock.Object);
+                Thread.Sleep(2000);
+            });
+            await SqlTestDb.DropDatabase(testDb.DatabaseName);
+        }
+
+        private static bool VerifySendedFeedback(bool result, bool expectedResult)
+        {
+            return result == expectedResult;
+        }
+
+        private RenameRequestParams InitRequestParams(string newName, string UrnOfObject)
```

Comment:
Doesn't seem like something that needs a helper function given that all it's doing is creating a single object

Reply:
I am not sure. Is it not redudant if I have to create the object in every test-method? For me personally I would say that it's more intuitive and saves also a lot of code lines.